src/binary-writer.cc

```diff
@@ -1188,8 +1189,17 @@ Result BinaryWriter::WriteModule() {
       Offset body_size_offset =
           WriteU32Leb128Space(leb_size_guess, "func body size (guess)");
       WriteFunc(func);
-      WriteFixupU32Leb128Size(body_size_offset, leb_size_guess,
+      auto func_start_offset = body_size_offset - last_section_payload_offset_;
+      auto func_end_offset = stream_->offset() - last_section_payload_offset_;
+      auto delta = WriteFixupU32Leb128Size(body_size_offset, leb_size_guess,
                               "FIXUP func body size");
+      if (current_reloc_section_ && delta != 0) {
```

Comment:
Why are the relocations only written at the end of the functions section? Can you explain the bug that this is fixing a bit more?